processing/src/test/java/org/apache/druid/query/aggregation/cardinality/CardinalityAggregatorTest.java

```diff
@@ -405,8 +411,27 @@ public void testAggregateValues()
     for (int i = 0; i < VALUES1.size(); ++i) {
       aggregate(selectorList, agg);
     }
-    Assert.assertEquals(NullHandling.replaceWithDefault() ? 7.0 : 6.0, (Double) valueAggregatorFactory.finalizeComputation(agg.get()), 0.05);
-    Assert.assertEquals(NullHandling.replaceWithDefault() ? 7L : 6L, rowAggregatorFactoryRounded.finalizeComputation(agg.get()));
+    if (NullHandling.ignoreNullsForStringCardinality()) {
```

Comment:
it would be useful to have a method for tests on `NullHandling` that allows explicitly configuring this value so you can test both modes (to be ultra safe, in a try/finally to revert it in case anything happens it doesn't leave static config in a funny state)